server/src/main/java/org/opensearch/index/similarity/SimilarityService.java

```diff
@@ -100,6 +100,7 @@ public final class SimilarityService extends AbstractIndexComponent {
             );
         });
         builtIn.put("BM25", (settings, version, scriptService) -> SimilarityProviders.createBM25Similarity(settings, version));
+        builtIn.put("LegacyBM25", (settings, version, scriptService) -> SimilarityProviders.createLegacyBM25Similarity(settings, version));
```

Comment:
@reta coming from https://github.com/opensearch-project/OpenSearch/pull/17306/files#r1948332363, this is where the `createLegacyBM25Similarity` is called

I have with the tested the following sample query
```
curl -X PUT "http://localhost:9200/test-index" -H 'Content-Type: application/json' -d'
{
  "settings": {
    "index": {
      "similarity": {
        "default": {
          "type": "LegacyBM25",
          "k1": 1.2,
          "b": 0.75
        }
      }
    }
  },
  "mappings": {
    "properties": {
      "content": {
        "type": "text"
      }
    }
  }
}
'
```

Reply:
thanks @prudhvigodithi , let's see what other folks thinks, should we keep it or not (preferably, not I believe)

Reply:
Its already been marked Deprecated at least since this [commit](https://github.com/opensearch-project/OpenSearch/commit/006c832c5fe8f509aa6285de90f2c7583b3dff35), so I think it would be fair to remove in 3.0.0.

Reply:
Thanks @reta and @cwperks, @msfroh @andrross can you please provide your thoughts.

Reply:
> Its already been marked Deprecated at least since this commit, so I think it would be fair to remove in 3.0.0.

The problem is that while that commit marked the Java class as `@deprecated`, it never provided a way for users to use the non-deprecated similarity. So, there was never a migration path. ☹️ 

IMO, we need to give folks an easy way to regain the default(/**only available**) 2.x behavior so that they can upgrade to 3.0.